src/org/traccar/events/GeofenceEventHandler.java

```diff
@@ -57,18 +59,24 @@ public GeofenceEventHandler() {
 
         Collection<Event> events = new ArrayList<>();
         for (long geofenceId : newGeofences) {
-            long calendarId = geofenceManager.getGeofence(geofenceId).getCalendarId();
-            if (calendarId == 0 || Context.getCalendarManager().getCalendar(calendarId) == null
-                    || Context.getCalendarManager().getCalendar(calendarId).checkMoment(position.getFixTime())) {
+            long calendarId = ((Geofence) geofenceManager.getById(geofenceId)).getCalendarId();
+            Calendar calendar = null;
+            if (calendarId != 0) {
```

Comment:
Can we use ternary expression here? If it's too long to fit on one line, please leave it as it is.